src/main/java/com/google/cloud/genomics/dataflow/utils/BAMDiff.java

```diff
@@ -0,0 +1,457 @@
+            h2.getSequenceDictionary().getSequence(sr1.getSequenceName()) : 
+            s2.get(i);
+        if (sr2 == null) {
+          error("Failed to find sequence " + sr1.getSequenceName() + " in " + BAMFile2);
+        }
+        ret = compareSequenceRecord(sr1, sr2, i + 1) && ret;
+      }
+    } else {
+      LOG.info("Comparing specified sequences in the headers");
+      for (String r : referencesToProcess) {
+        LOG.info("Comparing reference " + r);
+        ret = compareSequenceRecord(h1.getSequenceDictionary().getSequence(r),
+            h2.getSequenceDictionary().getSequence(r), -1) && ret;
+      }
+    }
+    return ret;
+}
+
+private boolean compareSequenceRecord(final SAMSequenceRecord sequenceRecord1, final SAMSequenceRecord sequenceRecord2, final int which) throws Exception {
+    if (!sequenceRecord1.getSequenceName().equals(sequenceRecord2.getSequenceName())) {
+        reportDifference(sequenceRecord1.getSequenceName(), sequenceRecord2.getSequenceName(),
+                "Name of sequence record " + which);
+        return false;
+    }
+    boolean ret = compareValues(sequenceRecord1.getSequenceLength(), sequenceRecord2.getSequenceLength(), "Length of sequence " +
+            sequenceRecord1.getSequenceName());
+    if (!options.ignoreSequenceProperties) {
+      ret = compareValues(sequenceRecord1.getSpecies(), sequenceRecord2.getSpecies(), "Species of sequence " +
+              sequenceRecord1.getSequenceName()) && ret;
+      ret = compareValues(sequenceRecord1.getAssembly(), sequenceRecord2.getAssembly(), "Assembly of sequence " +
+              sequenceRecord1.getSequenceName()) && ret;
+      ret = compareValues(sequenceRecord1.getAttribute("M5"), sequenceRecord2.getAttribute("M5"), "MD5 of sequence " +
+              sequenceRecord1.getSequenceName()) && ret;
+      ret = compareValues(sequenceRecord1.getAttribute("UR"), sequenceRecord2.getAttribute("UR"), "URI of sequence " +
+              sequenceRecord1.getSequenceName()) && ret;
+    }
+    return ret;
+}
+
+private <T> boolean compareValues(final T v1, final T v2, final String label) throws Exception {
+    if (v1 == null) {
+        if (v2 == null) {
+            return true;
+        }
+        reportDifference(v1, v2, label);
+        return false;
+    }
+    if (v2 == null) {
+        reportDifference(v1, v2, label);
+        return false;
+    }
+    if (!v1.equals(v2)) {
+        reportDifference(v1, v2, label);
+        return false;
+    }
+    return true;
+}
+
+private void reportDifference(final String s1, final String s2, final String label) throws Exception {
+    error(label + " differs.\n" +
+        BAMFile1 + ": " + s1 + "\n" + 
+        BAMFile2 + ": " + s2);
+}
+
+private void reportDifference(Object o1, Object o2, final String label) throws Exception {
+    if (o1 == null) {
+        o1 = "null";
+    }
+    if (o2 == null) {
+        o2 = "null";
+    }
+    reportDifference(o1.toString(), o2.toString(), label);
+}
+}
```

Comment:
Probably all the methods need just one indent as the curly brackets at the end align.